src/core/border_router/routing_manager.cpp

```diff
@@ -565,7 +565,7 @@ Error RoutingManager::PublishExternalRoute(const Ip6::Prefix &aPrefix, RoutePref
         LogWarn("Failed to publish external route %s: %s", aPrefix.ToString().AsCString(), ErrorToString(error));
     }
 
-    return error;
```

Comment:
Should `Get<NetworkData::Publisher>().PublishExternalRoute` return `kErrorNone` directly?

Reply:
Not sure if it would be a good idea. Since it is exposed as public API `otNetDataPublishExternalRoute` already (we should minimize changing public APIs). Also it can help us detect if/when same prefix is being published multiple times   e.g. from logging of error (so we detect it and see if it make sense)